code/main/media/MediaSession.brs

```diff
@@ -0,0 +1,266 @@
+' ********************** Copyright 2023 Adobe. All rights reserved. **********************
+ ' *
+ ' * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ ' * you may not use this file except in compliance with the License. You may obtain a copy
+ ' * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ ' *
+ ' * Unless required by applicable law or agreed to in writing, software distributed under
+ ' * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ ' * OF ANY KIND, either express or implied. See the License for the specific language
+ ' * governing permissions and limitations under the License.
+ ' *
+ ' *****************************************************************************************
+
+ ' ***************************** MODULE: MediaSession *******************************
+
+ function _adb_MediaSession(id as string, configurationModule as object, sessionConfig as object, edgeRequestQueue as object) as object
+     return {
+        _id = id,
```

Comment:
Is this clientSessionId? If so, can you rename this or add a comment?

Reply:
This is just an id passed by the sessionManager. Might be useful if we want to maintain a map. Apart from it this can be used to detect if session changed (auto restart etc..)

Reply:
As we discussed in the meeting, multiple "active" sessions may exist. When switching back to the previous screen, the previous client session may still be alive. So, it's possible to receive events from the previous session.  Let's add a map to cache the"active" session objects and use the `clientSessionId` as the uuid of the session object.
```
sessionObj = _adb_MediaSession(clientSessionId, configurationModule, sessionConfig, edgeRequestQueue)
_sessionMap[clientSessionId] = sessionObj
```
For `MediaSessionManager` exposed functions, using the `clientSessionId` to retrieve the corresponding session object if the `clientSessionId` is not matched with the `_activeSession`. 
```
queue: sub(clientSessionId as string, mediaHit as object)
    if m._activeSession <> invalid and m._activeSessionId = clientSessionId
        m._activeSession.process(mediaHit)
    end if
    if m._sessionMap.DoesExist(clientSessionId)
        session = m._sessionMap[clientSessionId]
        session.process(mediaHit)
    end if

end sub,
```

Thoughts?

Reply:
How will there be multiple media sessions, if we are ending any unended session when sessionstart is called? If the two screens use completely different instance then also we dont need to do anything as those instance will have their single media session.

Reply:
Let's discuss and implement it in another PR.